src/commonMain/kotlin/uuid.kt

```diff
@@ -158,6 +165,39 @@ public class Uuid(val uuid: ByteArray) {
             return Uuid(bytes)
         }
 
+        /**
+         * Parses a UUID from a String
+         *
+         * @param from The String, from which to deserialize the UUID
+         * @return a UUID, if the string is a valid UUID string or throws an [IllegalArgumentException]
+         */
+        fun fromString(from: String): Uuid {
+            if (from.length != UUID_STRING_LENGTH) {
+                throw IllegalArgumentException("Uuid string has invalid length: $from")
+            }
+            if (hyphenIndices.find { from[it] != '-' } != null) {
+                throw IllegalArgumentException("Uuid string has invalid format: $from (")
+            }
+
+            val bytes = ByteArray(UUID_BYTES)
+            var byte = 0
+            for (range in uuidCharRanges) {
```

Comment:
Thoughts on how to handle this code duplication better? 

I could get rid of specific error messages in `fromString()` and internally delegate to `parse()`. If `parse()` returns null, `fromString()` could throw an exception with a generic error message.

Reply:
Made a suggestion in another comment :)